setup.py

```diff
@@ -108,87 +105,22 @@ def run_tests(self):
             django.setup()
 
 
-cmdclass['test'] = PyTest
-
-
-class NpmInstall(Command):
-
-    def initialize_options(self):
-        pass
-
-    def finalize_options(self):
-        pass
-
-    def run(self):
-        self.spawn(['npm', 'install'])
-
-
-cmdclass['npm_install'] = NpmInstall
-
-
-class Grunt(Command):
-
-    sub_commands = [
-        ('npm_install', None)
-    ]
-
-    def initialize_options(self):
-        pass
-
-    def finalize_options(self):
-        pass
-
-    def run(self):
-        if not os.path.isdir('src'):
-            return
-        for command in self.get_sub_commands():
-            self.run_command(command)
-
-        self.spawn(['grunt', 'jst'])
-        self.spawn(['grunt', 'concat'])
-        self.spawn(['grunt', 'jshint', '--force'])
-        self.spawn(['grunt', 'copy:shapeEditor'])
-
-
-cmdclass['grunt'] = Grunt
-
-
-class Sdist(setuptools.command.sdist.sdist):
-
-    def run(self):
-        if os.path.isdir('src'):
-            self.run_command('grunt')
-        setuptools.command.sdist.sdist.run(self)
-
-
-cmdclass['sdist'] = Sdist
-
-
-class Install(setuptools.command.install.install):
-
-    def run(self):
-        if os.path.isdir('src'):
-            self.run_command('grunt')
-        setuptools.command.install.install.run(self)
-
-
-cmdclass['install'] = Install
-
-
-class Develop(setuptools.command.develop.develop):
-
-    sub_commands = setuptools.command.develop.develop.sub_commands + [
-        ('grunt', None)
-    ]
-
-    def run(self):
-        if os.path.isdir('src'):
-            for command in self.get_sub_commands():
-                self.run_command(command)
-        setuptools.command.develop.develop.run(self)
-
-
-cmdclass['develop'] = Develop
+def require_npm(command, strict=False):
+    """
+    Decorator to run NPM prerequisites
+    """
+    class WrappedCommand(command):
+        def run(self):
+            if strict or not os.path.isdir(
+                    'omero_figure/static/figure/js'):
```

Comment:
```suggestion
                    'omero_figure/static/figure'):
```
Looks like the files aren't under `js`:
```
        omero_figure/static/figure/figure.js
        omero_figure/static/figure/templates.js
```

Reply:
Actually that won't work, you'll need to check for the presence of the files not the directory